src/lib/modules/tests/test.js

```diff
@@ -337,30 +326,34 @@ class Test {
     });
   }
 
-  static getWeb3Contract(contract, web3) {
-    const newContract = new EmbarkJS.Blockchain.Contract({
-      abi: contract.abiDefinition,
-      address: contract.deployedAddress,
-      from: contract.deploymentAccount || web3.eth.defaultAccount,
-      gas: constants.tests.gasLimit,
-      web3: web3
-    });
+  getWeb3Contract(contract, web3, cb) {
+    this.events.request('runcode:eval', 'EmbarkJS', (err, embarkjs) => {
```

Comment:
It's a little weird that you have to eval EmbarkJS on each new contract. This could happen 20 times for a single test file.
Shouldn't we eval EmbarkJS at the start of the deploy instead?